### PR TITLE
Telemetry and Event Logging

### PR DESCRIPTION
**Pull request by Zsailer**
This is a port of jupyter/notebook#4752. 

This is an initial draft of telemetry/eventlogging in the Jupyter Server. This was inspired by #231 from @gclen. For now, this PR adds structured event logging to "contents" API, but we could easily draft event schemas for the kernels API after this lands.

Concurrently, I've drafted a large PR in the telemetry repo, jupyter/telemetry#46, for handling personal data. A major block for landing telemetry in jupyter_server has been the concern around the logging of personal data. Before we can merge something here, we need to have a sound story around how PII is should be handled. 

I will try to flesh out the details further in this PR when I have more time. In the meantime, I thought I'd leave this here for others to take a look at. (Thank you, @yuvipanda, for the initial draft of this PR!).

### CODE REVIEW

**Comment by ellisonbg** [comment]
Going to dive into this

File: jupyter_server/services/eventlog/handlers.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import json
+
+from tornado import web
+
+from notebook.utils import url_path_join
```
Comment (Zsailer) [inline jupyter_server/services/eventlog/handlers.py:5]: update to `jupyter_server`

**Comment by WaterKnight1998** [comment]
@Zsailer any updates?

**Comment by Zsailer** [comment]
Closing in favor of #352.

**Pull request closed without merging**